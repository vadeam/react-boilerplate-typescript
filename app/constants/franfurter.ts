export const DEFAULT_BASE_CURRENCY = 'EUR'
