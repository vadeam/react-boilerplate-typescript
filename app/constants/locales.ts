export const LOCALE_DE = 'de'
export const LOCALE_EN = 'en'
export const DEFAULT_LOCALE = LOCALE_EN

export const appLocales: Locales[] = [LOCALE_EN, LOCALE_DE]
export const CUSTOM_LOCALE_STORAGE_KEY = 'custom-locale'
