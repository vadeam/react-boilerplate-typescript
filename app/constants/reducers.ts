export const REDUCER_KEY_APP = 'app'
export const REDUCER_KEY_LANGUAGE = 'language'
export const REDUCER_KEY_ROUTER = 'router'
export const REDUCER_KEY_HOME = 'home'
// for testing purposes
export const REDUCER_KEY_TEST = 'test'
