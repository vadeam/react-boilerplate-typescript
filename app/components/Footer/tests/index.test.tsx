import React from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import { LOCALE_EN } from '../../../constants/locales'
import configureStore from '../../../configureStore'
import history from '../../../utils/history'
import Footer from '../index'

describe('<Footer />', () => {
  let store

  beforeAll(() => {
    store = configureStore({}, history)
  })

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider store={store}>
          <IntlProvider locale={LOCALE_EN}>
            <Footer />
          </IntlProvider>
        </Provider>,
      )
      .toJSON()

    expect(renderedComponent).toMatchSnapshot()
  })
})
