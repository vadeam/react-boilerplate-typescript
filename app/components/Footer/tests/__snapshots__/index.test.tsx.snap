// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
<footer
  className="sc-hKgJUU lkVlfl"
>
  <section>
    This project is licensed under the MIT license.
  </section>
  <section>
    <div
      className="sc-dlfnuX kMMaeO"
    >
      <select
        className="sc-gsTEea eEhauo"
        onChange={[Function]}
        value="en"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="de"
        >
          de
        </option>
      </select>
    </div>
  </section>
  <section>
    
      Made with love by 
    <a
      className="sc-bdfBQB limALa"
      href="https://twitter.com/mxstbr"
    >
      Max Stoiber
    </a>
    .
    
  </section>
</footer>
`;
